Upgrade docs (#16)
bioio-devs/bioio: docs/OVERVIEW.md

img.physical_pixel_sizes.Y  # returns the Y dimension pixel size as found in the metadata
img.physical_pixel_sizes.X  # returns the X dimension pixel size as found in the metadata
```

### Xarray Coordinate Plane Attachment

If `bioio` finds coordinate information for the spatial-temporal dimensions of
the image in metadata, you can use
[xarray](http://xarray.pydata.org/en/stable/index.html) for indexing by coordinates.

```python
from bioio import BioImage

# Get a BioImage object
img = BioImage("my_file.ome.tiff")

# Get the first ten seconds (not frames)
first_ten_seconds = img.xarray_data.loc[:10]  # returns an xarray.DataArray

# Get the first ten major units (usually micrometers, not indices) in Z
first_ten_mm_in_z = img.xarray_data.loc[:, :, :10]

# Get the first ten major units (usually micrometers, not indices) in Y
first_ten_mm_in_y = img.xarray_data.loc[:, :, :, :10]

# Get the first ten major units (usually micrometers, not indices) in X
first_ten_mm_in_x = img.xarray_data.loc[:, :, :, :, :10]
```

See `xarray`
["Indexing and Selecting Data" Documentation](http://xarray.pydata.org/en/stable/indexing.html)
for more information.

> ### Cloud IO Support

Reviewer: was just looking at the live version and for some reason this got rendered with `#id2` instead of `#cloud-io-support`.  There is a link to it near the top of the document which now seems broken.

Author: weird, but adjusted

from bioio import BioImage

BioImage("my_file.czi").save("my_file.ome.tiff")
```

**Note:** By default `BioImage` will generate only a portion of metadata to pass
along from the reader to the OME model. This function currently does not do a full
metadata translation.

For finer grain customization of the metadata, scenes, or if you want to save an array
as an OME-TIFF, the writer class can also be used to customize as needed.

```python
import numpy as np
from bioio.writers import OmeTiffWriter

image = np.random.rand(10, 3, 1024, 2048)
OmeTiffWriter.save(image, "file.ome.tif", dim_order="ZCYX")
```

See
[OmeTiffWriter documentation](https://bioio-devs.github.io/bioio/bioio.writers.html)
for more details.

#### Other Writers

In most cases, `BioImage.save` is usually a good default but there are other image
writers available. For more information, please refer to
[our writers documentation](https://bioio-devs.github.io/bioio/bioio.writers.html).

## Development

See our
[developer resources](https://bioio-devs.github.io/bioio/developer_resources.html)
for information related to developing the code.

## Citation

If you find `bioio` useful, please cite this repository as:

> > Eva Maxfield Brown, Dan Toloudis, Jamie Sherman, Madison Swain-Bowden, Talley Lambert, Sean Meharry, Brian Whitney, AICSImageIO Contributors (2023). AICSImageIO: Image Reading, Metadata Conversion, and Image Writing for Microscopy Images in Pure Python [Computer software]. GitHub. https://github.com/AllenCellModeling/aicsimageio

Reviewer: ```suggestion
> Eva Maxfield Brown, Dan Toloudis, Jamie Sherman, Madison Swain-Bowden, Talley Lambert, Sean Meharry, Brian Whitney, AICSImageIO Contributors (2023). BioIO: Image Reading, Metadata Conversion, and Image Writing for Microscopy Images in Pure Python [Computer software]. GitHub. https://github.com/AllenCellModeling/aicsimageio
```

> > Eva Maxfield Brown, Dan Toloudis, Jamie Sherman, Madison Swain-Bowden, Talley Lambert, Sean Meharry, Brian Whitney, AICSImageIO Contributors (2023). AICSImageIO: Image Reading, Metadata Conversion, and Image Writing for Microscopy Images in Pure Python [Computer software]. GitHub. https://github.com/AllenCellModeling/aicsimageio

Reviewer: ```suggestion
> Eva Maxfield Brown, Dan Toloudis, Jamie Sherman, Madison Swain-Bowden, Talley Lambert, Sean Meharry, Brian Whitney, AICSImageIO Contributors (2023). BioIO: Image Reading, Metadata Conversion, and Image Writing for Microscopy Images in Pure Python [Computer software]. GitHub. https://github.com/bioio-devs/bioio
```

# BioIO

[![Build Status](https://github.com/bioio-devs/bioio/actions/workflows/ci.yml/badge.svg)](https://github.com/bioio-devs/bioio/actions)

> [![Documentation](https://github.com/bioio-devs/bioio/actions/workflows/docs.yml/badge.svg)](https://bioio-devs.github.io/bioio)

Reviewer: I don't know if we need the documentation and build status badges here in this doc.

[![PyPI version](https://badge.fury.io/py/bioio.svg)](https://badge.fury.io/py/bioio)
[![License](https://img.shields.io/badge/License-BSD%203--Clause-blue.svg)](https://opensource.org/licenses/BSD-3-Clause)

> [![Python 3.6](https://img.shields.io/badge/python-3.9,3.10,3.11-blue.svg)](https://www.python.org/downloads/release/python-360/)

Reviewer: ```suggestion
[![Python 3.9+](https://img.shields.io/badge/python-3.9,3.10,3.11-blue.svg)](https://www.python.org/downloads/release/python-390/)
```

Reviewer: Future/todo: Should we add 3.12 ?

Image Reading, Metadata Conversion, and Image Writing for Microscopy Images in Pure Python

> [_Click here to view all open issues in bioio-devs organization at once_](https://github.com/search?q=user%3Abioio-devs+is%3Aissue+is%3Aopen&type=issues&ref=advsearch)

Reviewer: Nit - should this link be in the main docs?  I don't deny it's very useful.

Author: Ehhh maybe not, I'll remove

from bioio import BioImage

BioImage("my_file.czi").save("my_file.ome.tiff")
```

**Note:** By default `BioImage` will generate only a portion of metadata to pass
along from the reader to the OME model. This function currently does not do a full
metadata translation.

For finer grain customization of the metadata, scenes, or if you want to save an array
as an OME-TIFF, the writer class can also be used to customize as needed.

```python
import numpy as np
from bioio.writers import OmeTiffWriter

image = np.random.rand(10, 3, 1024, 2048)
OmeTiffWriter.save(image, "file.ome.tif", dim_order="ZCYX")
```

See
[OmeTiffWriter documentation](https://bioio-devs.github.io/bioio/bioio.writers.html)
for more details.

#### Other Writers

In most cases, `BioImage.save` is usually a good default but there are other image
writers available. For more information, please refer to
[our writers documentation](https://bioio-devs.github.io/bioio/bioio.writers.html).

## Development

See our
[developer resources](https://bioio-devs.github.io/bioio/developer_resources.html)
for information related to developing the code.

## Citation

If you find `bioio` useful, please cite this repository as:

> > Eva Maxfield Brown, Dan Toloudis, Jamie Sherman, Madison Swain-Bowden, Talley Lambert, Sean Meharry, Brian Whitney, AICSImageIO Contributors (2023). BioIO: Image Reading, Metadata Conversion, and Image Writing for Microscopy Images in Pure Python [Computer software]. GitHub. https://github.com/bioio-devs/bioio

Reviewer: still not sure if we need to change the contributors list or just completely preserve the entire AICSImageIO lineage.  It still says `AICSImageIO Contributors` and maybe we should either add `BioIO Contributors` or replace.  Also kinda wondering if this conversation needs to involve AICS governance.
